### DISCUSSION TITLE
Purpose of the components

### DISCUSSION THREAD

**Discussion by ivanov-v** [ariakit/ariakit#1444] [Questions] [answered]
Hello. Do you plan to make detailed documentation for each component?
For example, now it is not clear why the Collection component is needed. Or do I need to use Command instead of Button and when.
Thanks.

**Comment by diegohaz** [comment, marked as answer]
Yes, the docs will have detailed information for each component.

Collection and Command are abstract components that provide features shared by other abstract/concrete components.

Collection keeps a list of CollectionItem's data in the state (`collection.items`) in the exact order they're rendered in the DOM, and observes the DOM for any change in their order. This is used by Composite (which is another abstract component used by Combobox, Select, Menu, etc.) and Form.

Command represents any clickable element: Button, Checkbox, Radio, etc.

**Reply by ivanov-v** [reply]
Cool! Thanks)
